R/combine_checkboxes.R

```diff
@@ -272,19 +282,55 @@ replace_true <- function(col, col_name, metadata, raw_or_label) {
 #' @param data_tbl The data tibble from the original supertibble
 #' @param .new_value The new column values made by [combine_checkboxes()]
 #' @inheritParams combine_checkboxes
-convert_checkbox_vals <- function(metadata, .new_value, data_tbl, raw_or_label, multi_value_label, values_fill) {
-  tibble(
-    !!.new_value := rowSums(!is.na(data_tbl[names(data_tbl) %in% metadata$field_name]))
-  ) %>%
+convert_checkbox_vals <- function(metadata,
+                                  .new_value,
+                                  data_tbl,
+                                  raw_or_label,
+                                  multi_value_label,
+                                  values_fill,
+                                  multi_value_sep) {
+  use_multi_value_label <- !is.null(multi_value_label)
+  multi_value_label <- if (use_multi_value_label) multi_value_label else NA_character_
+
+  out <- data_tbl %>%
+    unite(
+      ".combined",
+      any_of(metadata$field_name),
+      sep = multi_value_sep,
+      na.rm = TRUE,
+      remove = FALSE
+    ) %>%
     mutate(
-      !!.new_value := case_when(. > 1 ~ multi_value_label,
-        . == 1 ~ coalesce(!!!data_tbl[, names(data_tbl) %in% metadata$field_name]),
+      .rowsum = rowSums(!is.na(data_tbl[names(data_tbl) %in% metadata$field_name])),
+      !!.new_value := case_when(
+        .rowsum > 1 & !use_multi_value_label ~ .combined,
+        .rowsum > 1 & use_multi_value_label ~ multi_value_label,
+        .rowsum == 1 ~ coalesce(!!!data_tbl[, names(data_tbl) %in% metadata$field_name]),
         .default = values_fill
-      ),
-      !!.new_value := factor(!!sym(.new_value),
-        levels = c(metadata[[raw_or_label]], multi_value_label, values_fill)
       )
     )
+
+  # If using multi_value_label, preserve factor levels
+  # Otherwise leave as character with concatenated values
+  if (use_multi_value_label) {
+    lvl_vec <- unique(c(
+      metadata[[raw_or_label]],
+      multi_value_label,
+      out[[.new_value]],
```

Comment:
This _should be_ redundant because all values in the data need to come from one of the other options.

Reply:
You mean just `out[[.new_value]]`?

Reply:
Yes!